core/src/main/java/io/knotx/forms/core/domain/FormEntity.java

```diff
@@ -48,14 +48,26 @@
   private Map<String, String> signalToUrl;
 
   public static FormEntity from(Fragment fragment, FormsKnotOptions options) {
+    FormEntity result = null;
     Element scriptDocument = FragmentContentExtractor.unwrapFragmentContent(fragment);
-    return new FormEntity()
-        .fragment(fragment)
-        .identifier(getFormIdentifier(fragment))
-        .adapterParams(getAdapterParams(scriptDocument))
-        .adapter(getAdapterMetadata(options, getAdapterName(fragment, scriptDocument)))
-        .signalToUrlMapping(getSignalToUrlMapping(scriptDocument));
-
+    try {
+      result = new FormEntity()
+          .fragment(fragment)
+          .identifier(getFormIdentifier(fragment))
+          .adapterParams(getAdapterParams(scriptDocument))
+          .adapter(getAdapterMetadata(options, getAdapterName(fragment, scriptDocument)))
+          .signalToUrlMapping(getSignalToUrlMapping(scriptDocument));
+    } catch (Exception e) {
```

Comment:
We have the method `io.knotx.forms.core.FormsKnotProxy#processRequest` that wraps processing with RX Sindle. So the exception handling can be moved from here to `io.knotx.forms.core.FormsKnotProxy#FormsKnotProxy`. We would lose `io.knotx.forms.core.domain.FormsFactory#areUnique` validation as far as we would process each fragment in stream.

So the `FormsKnotProxy` will contain:
```
  @Override
  public Single<KnotContext> processRequest(final KnotContext knotContext) {

    return Single.just(knotContext)
        .flattenAsObservable(KnotContext::getFragments)
        .filter(f -> f.knots().stream().anyMatch(id -> id.startsWith(
            FormConstants.FRAGMENT_KNOT_PREFIX)))
        .map(f -> FormEntity.from(f, options))
        .onErrorReturnItem(form -> handleFallack(form))
        .toList()
        .flatMap(forms -> {
          if (knotContext.getClientRequest().getMethod() == HttpMethod.GET) {
            return Single.just(handleGetMethod(forms, knotContext));
          } else {
            FormEntity current = currentForm(forms, knotContext);
            return callFormsAdapter(knotContext, current)
                .map(response -> processAdapterResponse(knotContext, forms, current, response));
          }
        })
        .onErrorReturn(error -> processError(knotContext, error));
  }

  private FormEntity handleFallack(Throwable form) {
    return null;
  }
```
and `FormsFactory` can be removed.

Reply:
done

Reply:
Good job!